### ISSUE TITLE
Is it possible to iterate over a Selection concurrently?

### ISSUE THREAD

**Issue by atye** [PuerkitoBio/goquery#351]
I have some code that looks like this:

```
doc.Find("div.class").EachWithBreak(func(_ int, s *goquery.Selection) bool {
	err := doSomeWork(s)
	if err != nil {
		return false
	}

	return true
})
```

Is there a way to use `EachWithBreak` or `Each` concurrently so that I can work on each Selection Node at the same time? I could probably use go routines and channels but it feels like handling the `bool` return could get messy. I could iterate over the Selection.Nodes but then I'd lose some of the Selection functionality I'm using elsewhere like `AttrOr`, `Text`, etc.

Basically, I want to process each Selection concurrently but still use the WithBreak functionality if there is an error.

**Comment by atye** [comment]
I'm realizing that this question isn't really a goquery issue.
